### DISCUSSION TITLE
How to custom branch name for non-default branches

### DISCUSSION THREAD

**Discussion by greyli-dell** [renovatebot/renovate#30031] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted, Renovate 37.415.0

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub Enterprise 3.12.4

### Please tell us more about your question or problem

Hi team,

When multiple branches are enabled with the "baseBranches" option, the branch name for non-default branch will become:

```
{{ branchPrefix }}{{ base branch name}}{{additionalBranchPrefix}}{{ branchTopic }}
```

For example, to update the dep in the `dev-branch` branch, the created branch will be `my-prefix/dev-branch-pypi-aiohttp-xxx`, where the `my-prefix/` part is the branch prefix, and the `pypi-aiohttp-xxx` is the branch topic part.

I can customize the branch prefix and topic with the corresponding `branchPrefix` and `branchTopic` config options. How could I update the `base branch name` part? I want to replace the "/" to "-" in it.

I also try to set the `branchName` explicitly:

```
branchName: "{{{branchPrefix}}}{{{additionalBranchPrefix}}}{{{branchTopic}}}",
branchPrefix: "pvt/",                                                                                                   
additionalBranchPrefix: "",                                                                                             
branchTopic: "{{{replace '/' '-' depNameSanitized}}}-...",
branchNameStrict: true
```

But it will still contain the branch name info between `branchPrefix` and `branchTopic`.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Can you give an example of an undesired branch name you're seeing and the desired branch name?

**Reply by greyli-dell** [reply]
Sure. With `baseBranches` enabled for the `$default` and `rel/1.1.x` branch, the dep update branch created for `rel/1.1.x` will be:

```
prefix/rel/1.1.x-pypi-requests-vulnerability
```

Since our branch protection rule prevents using multiple slashes in the branch name, I hope I can custom the base branch name part, so I can get a branch name like this:

```
prefix/rel-1.1.x-pypi-requests-vulnerability
```

**Reply by greyli-dell** [reply]
Hi @rarkins, any updates?

**Reply by rarkins** [reply, marked as answer]
I'm not able to think of a way to achieve this currently, so created an issue for it: https://github.com/renovatebot/renovate/issues/30266
